### PR TITLE
Moving pillar to Go 12 and modules

### PR DESCRIPTION
**Pull request by rvs**

### CODE REVIEW

File: pkg/pillar/Dockerfile.in [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
 # The only one supported so far is cross compiling for aarch64 on x86
+ENV GOFLAGS=-mod=vendor
+ENV GO111MODULE=on
 ENV CGO_ENABLED=1
 ARG GOARCH=
 ARG CROSS_GCC=https://musl.cc/aarch64-linux-musleabi-cross.tgz
 RUN [ -z "$GOARCH" ] || (cd / ; apk add --no-cache wget && wget -O - $CROSS_GCC | tar xzvf -)
 
-ADD ./  /go/src/github.com/zededa/eve/pkg/pillar/
+ADD ./  /pillar/
```
Comment (deitch) [inline pkg/pillar/Dockerfile.in:23]: We should split this into multiple steps:

```dockerfile
WORKDIR /pillar/
COPY go.mod .
COPY go.sum .
RUN go mod download
COPY . .
```

This prevents the entire go cache from being busted if we change a single file.

**Review by deitch — CHANGES_REQUESTED** [review]
This looks pretty good, is clean, just pillar.

The only requested change is to reduce the overhead of a single file busting the entire cache.

File: pkg/pillar/Dockerfile.in (hunk 1 above)
Comment (rvs) [reply pkg/pillar/Dockerfile.in:23]: Well, it isn't worse than what there before -- so I suggest we fix it in a follow up commit.

Comment (deitch) [reply pkg/pillar/Dockerfile.in:23]: Sounds good.

**Review by deitch — APPROVED** [review]

**Pull request merged**